101-dtl-create-lab/Cmdlets.DevTestLab.ps1

```diff
@@ -403,6 +516,92 @@ function New-AzureDtlLab
 
 ##################################################################################################
 
+function New-AzureDtlVMTemplate
+{
+    <#
+        .SYNOPSIS
+        Creates a new (or updates an existing) virtual machine template.
+
+        .DESCRIPTION
+        The New-AzureDtlVMTemplate cmdlet creates a new VM template from an existing VM.
+        - The new VM template is created in the same lab as the VM.
+        - If a VM template with the same name already exists in the lab, then it simply updates it.
+
+        .EXAMPLE
+        $lab = $null
+
+        $vm = Get-AzureDtlVirtualMachine -VMName "MyVM1"
+        New-AzureDtlVMTemplate -ExistingVM $lab -VMTemplateName "MyVMTemplate1" -VMTemplateDescription "MyDescription"
+
+        Creates a new VM Template "MyVMTemplate1" from the VM "MyVM1".
+
+        .INPUTS
+        None.
+    #>
+    [CmdletBinding()]
+    Param(
+        [ValidateNotNull()]
+        # An existing VM from which the new VM template will be created (please use the Get-AzureDtlVirtualMachine cmdlet to get this lab object).
+        $VM,
+
+        [ValidateNotNullOrEmpty()]
+        [string]
+        # Name of the new VM template to create.
+        $VMTemplateName,
+
+        [ValidateNotNull()]
+        [string]
+        # Details about the new VM template being created.
+        $VMTemplateDescription = ""
+    )
+
+    PROCESS
+    {
+        Write-Verbose $("Processing cmdlet '" + $PSCmdlet.MyInvocation.InvocationName + "', ParameterSet = '" + $PSCmdlet.ParameterSetName + "'")
+
+        # Folder location of VM creation script, the template file and template parameters file.
+        $VMTemplateCreationTemplateFile = Join-Path $PSScriptRoot -ChildPath "..\201-dtl-create-vmtemplate\azuredeploy.json" -Resolve
+
+        # Pre-condition check to ensure the RM template file exists.
+        if ($false -eq (Test-Path -Path $VMTemplateCreationTemplateFile))
+        {
+            Write-Error $("The RM template file could not be located at : '" + $VMTemplateCreationTemplateFile + "'")
+        }
+        else
+        {
+            Write-Verbose $("The RM template file was located at : '" + $VMTemplateCreationTemplateFile + "'")
+        }
+
+        # Get the lab that contains the source VM
+        $lab = GetLabFromVM_Private -VM $VM
+
+        if ($null -eq $lab)
+        {
+            Write-Error $("Unable to find lab '" + $LabName + "'")
+        }
+        else
+        {
+            # encode the VM template name
+            $VMTemplateNameEncoded = $VMTemplateName.Replace(" ", "%20")
+

```

Comment:
@ankurmittal: This is how we're temporarily working around the URL encoding issue for the VM template name.